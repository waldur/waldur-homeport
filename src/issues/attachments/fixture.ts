import { Attachment } from './types';

export const attachment: Attachment = {
  created: '2018-01-11T19:41:50.451396Z',
  file_size: 64145,
  file: 'https://example.com/media/support_attachments/panda.jpg',
  file_name: 'panda.jpg',
  mime_type: 'image/jpg',
  thumbnail:
    'https://example.com/media/support_attachments_thumbnails/panda.jpg',
  issue:
    'https://example.com/api/support-issues/784fd0b3502849f6bd613d574e06b1e4/',
  issue_key: 'WAL-164',
  url: 'https://example.com/api/support-attachments/30a44649ee2d4a67ba1f17938a8a5f6e/',
  uuid: '30a44649ee2d4a67ba1f17938a8a5f6e',
};
