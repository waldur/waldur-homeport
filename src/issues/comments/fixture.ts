import { Comment, Issue } from './types';

export const comment: Comment = {
  author_name: 'author_name',
  author_user: 'author_user',
  author_uuid: 'author_uuid',
  author_email: 'author_emai',
  backend_id: 'backend_id',
  created: 'created',
  description: 'description',
  is_public: true,
  issue: 'issue',
  issue_key: 'issue_key',
  url: 'url',
  uuid: 'uuid',
};

export const issue: Issue = {
  url: 'https://example.com/en-US/docs/Web/CSS/pointer-events',
};

export const users = {
  staff: {
    uuid: 'author_uuid_staff',
    is_staff: true,
  },
  same: {
    uuid: comment.author_uuid,
    is_staff: false,
  },
  different: {
    uuid: 'author_uuid_different',
    is_staff: false,
  },
};
