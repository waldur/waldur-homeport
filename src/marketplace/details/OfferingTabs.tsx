import * as React from 'react';

import { FormattedHtml } from '@waldur/core/FormattedHtml';
import { translate } from '@waldur/i18n';
import { Section, Offering } from '@waldur/marketplace/types';

import { AttributesTable } from './attributes/AttributesTable';
import { OfferingTab } from './OfferingTabsComponent';
import { OverviewTab } from './OverviewTab';
import { ScreenshotsTab } from './ScreenshotsTab';

interface OfferingTabsProps {
  sections: Section[];
  offering: Offering;
}

export const getTabs = (props: OfferingTabsProps): OfferingTab[] => {
  const attributes = props.offering.attributes;
  const filterSection = (section: Section) =>
    section.attributes.some(attr =>
      props.offering.attributes.hasOwnProperty(attr.key),
    );
  const sections = props.sections.filter(filterSection);

  const basicSections = sections.filter(s => s.is_standalone === false);
  const standaloneSections = sections.filter(s => s.is_standalone === true);

  let tabs = [
    {
      visible: !!props.offering.full_description,
      title: translate('Description'),
      component: () => <OverviewTab offering={props.offering} />,
    },
    {
      visible: !!props.offering.terms_of_service,
      title: translate('Terms of service'),
      component: () => <FormattedHtml html={props.offering.terms_of_service} />,
    },
    {
      visible: basicSections.length > 0,
      title: translate('Features'),
      component: () => (
        <AttributesTable attributes={attributes} sections={basicSections} />
      ),
    },
    {
      visible: props.offering.screenshots.length > 0,
      title: translate('Screenshots'),
      component: () => (
        <ScreenshotsTab screenshots={props.offering.screenshots} />
      ),
    },
  ];

  standaloneSections.forEach(section => {
    tabs.push({
      visible: true,
      title: section.title,
      component: () => (
        <AttributesTable attributes={attributes} sections={[section]} />
      ),
    });
  });
  tabs = tabs.filter(tab => tab.visible);
  return tabs;
};
