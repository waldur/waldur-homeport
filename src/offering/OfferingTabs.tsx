import * as React from 'react';
import * as Col from 'react-bootstrap/lib/Col';
import * as Panel from 'react-bootstrap/lib/Panel';
import * as Row from 'react-bootstrap/lib/Row';
import * as Tab from 'react-bootstrap/lib/Tab';
import * as Tabs from 'react-bootstrap/lib/Tabs';
import { connect } from 'react-redux';

import { withTranslation, TranslateProps } from '@waldur/i18n';
import { IssueCommentsContainer } from '@waldur/issues/comments/IssueCommentsContainer';
import { ResourceOrderItems } from '@waldur/marketplace/orders/item/list/ResourceOrderItems';
import { isVisible } from '@waldur/store/config';

import { OfferingEvents } from './OfferingEvents';
import { OfferingSummaryTab } from './OfferingSummaryTab';
import { OracleReport } from './OracleReport';
import { OracleSnapshots } from './OracleSnapshots';
import { Offering } from './types';
import { isOracleOffering } from './utils';

interface OwnProps {
  offering: Offering;
  summary: string;
}

interface StateProps {
  showComments: boolean;
}

type OfferingTabsProps = OwnProps & TranslateProps & StateProps;

export const PureOfferingTabs: React.FC<OfferingTabsProps> = (props) => {
  const issue = {
    uuid: props.offering.issue_uuid,
    url: props.offering.issue,
  };
  const showOracleReport =
    isOracleOffering(props.offering) && props.offering.report;
  const Summary = showOracleReport ? (
    <Row>
      <Col md={6}>
        <div className="m-b-md">
          <OfferingSummaryTab
            offering={props.offering}
            summary={props.summary}
          />
        </div>
        <Panel>
          <Panel.Heading>
            <Panel.Title>{props.translate('Snapshots')}</Panel.Title>
          </Panel.Heading>
          <Panel.Body>
            <OracleSnapshots report={props.offering.report} />
          </Panel.Body>
        </Panel>
      </Col>
      <Col md={6}>
        <OracleReport report={props.offering.report} />
      </Col>
    </Row>
  ) : (
    <OfferingSummaryTab offering={props.offering} summary={props.summary} />
  );
  return (
    <Tabs
      mountOnEnter
      unmountOnExit
      defaultActiveKey="summary"
      id="offeringSummary"
      animation={false}
    >
      <Tab title={props.translate('Summary')} eventKey="summary">
        <div className="m-t-sm">{Summary}</div>
      </Tab>
      <Tab title={props.translate('Audit log')} eventKey="events">
        <div className="m-t-sm">
          <OfferingEvents offering={props.offering} />
        </div>
      </Tab>
      {props.showComments && (
        <Tab title={props.translate('Comments')} eventKey="comments">
          <div className="m-t-sm">
            <IssueCommentsContainer issue={issue} renderHeader={false} />
          </div>
        </Tab>
      )}
      {props.offering.marketplace_resource_uuid && (
        <Tab title={props.translate('Order items')} eventKey="order-items">
          <div className="m-t-sm">
            <ResourceOrderItems
              resource_uuid={props.offering.marketplace_resource_uuid}
            />
          </div>
        </Tab>
      )}
    </Tabs>
  );
};

const connector = connect<StateProps, {}, OwnProps>((state, ownProps) => ({
  showComments:
    isVisible(state, 'offering.comments') &&
    Boolean(ownProps.offering.issue && ownProps.offering.issue_key),
}));

export const OfferingTabs = withTranslation(
  connector(PureOfferingTabs),
) as React.ComponentType<OwnProps>;
