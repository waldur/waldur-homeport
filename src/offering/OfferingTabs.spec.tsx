import { mount } from 'enzyme';
import * as React from 'react';
import { Provider } from 'react-redux';
import configureStore from 'redux-mock-store';

import { OfferingTabs } from './OfferingTabs';
import { OFFERING } from './OfferingTabs.fixture';

const renderComponent = () => {
  const mockStore = configureStore();
  const store = mockStore({ config: { featuresVisible: true } });
  const wrapper = mount(
    <Provider store={store}>
      <OfferingTabs offering={OFFERING} summary="" />
    </Provider>,
  );
  return wrapper.html();
};

describe('OfferingTabs', () => {
  it('renders tabs for Oracle report and snapshots', () => {
    const html = renderComponent();
    expect(html).toMatchSnapshot();
  });
});
