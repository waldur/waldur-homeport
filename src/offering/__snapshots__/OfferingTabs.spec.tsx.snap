// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`OfferingTabs renders tabs for Oracle report and snapshots 1`] = `
"<div id=\\"offeringSummary\\"><ul role=\\"tablist\\" class=\\"nav nav-tabs\\"><li role=\\"presentation\\" class=\\"active\\"><a id=\\"offeringSummary-tab-summary\\" role=\\"tab\\" aria-controls=\\"offeringSummary-pane-summary\\" aria-selected=\\"true\\" href=\\"#\\">Summary</a></li><li role=\\"presentation\\" class=\\"\\"><a id=\\"offeringSummary-tab-events\\" role=\\"tab\\" aria-controls=\\"offeringSummary-pane-events\\" tabindex=\\"-1\\" aria-selected=\\"false\\" href=\\"#\\">Audit log</a></li><li role=\\"presentation\\" class=\\"\\"><a id=\\"offeringSummary-tab-comments\\" role=\\"tab\\" aria-controls=\\"offeringSummary-pane-comments\\" tabindex=\\"-1\\" aria-selected=\\"false\\" href=\\"#\\">Comments</a></li></ul><div class=\\"tab-content\\"><div id=\\"offeringSummary-pane-summary\\" aria-labelledby=\\"offeringSummary-tab-summary\\" role=\\"tabpanel\\" aria-hidden=\\"false\\" class=\\"tab-pane active\\"><div class=\\"m-t-sm\\"><div class=\\"row\\"><div class=\\"col-md-6\\"><div class=\\"m-b-md\\"><div class=\\"panel panel-default\\"><div class=\\"panel-heading\\"><div class=\\"panel-title\\">Description</div></div><div class=\\"panel-body\\"><div>Tenants: ---<br>Additional Info: <br>- Organization: ActiveSys<br>- Project: SaaS<br></div></div></div></div><div class=\\"panel panel-default\\"><div class=\\"panel-heading\\"><div class=\\"panel-title\\">Snapshots</div></div><div class=\\"panel-body\\"><div class=\\"table-responsive\\"><table class=\\"table\\"><thead><tr><th>Name</th><th>Copy of</th><th>Creation time</th></tr></thead><tbody><tr><td>ora-tstardb-asm01.170927</td><td>ora-tstardb-asm01</td><td>2017-09-27 23:30:01 GST</td></tr><tr><td>ora-tstardb-asm01.170928</td><td>ora-tstardb-asm01</td><td>2017-09-28 23:30:01 GST</td></tr><tr><td>ora-tstardb-asm01.170929</td><td>ora-tstardb-asm01</td><td>2017-09-29 23:30:01 GST</td></tr><tr><td>ora-tstardb-asm01.170930</td><td>ora-tstardb-asm01</td><td>2017-09-30 23:30:02 GST</td></tr><tr><td>ora-tstardb-asm01.171001</td><td>ora-tstardb-asm01</td><td>2017-10-01 23:30:01 GST</td></tr><tr><td>ora-tstardb-asm01.171002</td><td>ora-tstardb-asm01</td><td>2017-10-02 23:30:02 GST</td></tr><tr><td>ora-tstardb-asm01.171003</td><td>ora-tstardb-asm01</td><td>2017-10-03 23:30:01 GST</td></tr><tr><td>ora-tstardb-asm02.170927</td><td>ora-tstardb-asm02</td><td>2017-09-27 23:30:01 GST</td></tr><tr><td>ora-tstardb-asm02.170928</td><td>ora-tstardb-asm02</td><td>2017-09-28 23:30:01 GST</td></tr><tr><td>ora-tstardb-asm02.170929</td><td>ora-tstardb-asm02</td><td>2017-09-29 23:30:01 GST</td></tr><tr><td>ora-tstardb-asm02.170930</td><td>ora-tstardb-asm02</td><td>2017-09-30 23:30:02 GST</td></tr><tr><td>ora-tstardb-asm02.171001</td><td>ora-tstardb-asm02</td><td>2017-10-01 23:30:01 GST</td></tr><tr><td>ora-tstardb-asm02.171002</td><td>ora-tstardb-asm02</td><td>2017-10-02 23:30:02 GST</td></tr><tr><td>ora-tstardb-asm02.171003</td><td>ora-tstardb-asm02</td><td>2017-10-03 23:30:01 GST</td></tr><tr><td>ora-tstardb-sys.170927</td><td>ora-tstardb-sys</td><td>2017-09-27 23:30:01 GST</td></tr><tr><td>ora-tstardb-sys.170928</td><td>ora-tstardb-sys</td><td>2017-09-28 23:30:01 GST</td></tr><tr><td>ora-tstardb-sys.170929</td><td>ora-tstardb-sys</td><td>2017-09-29 23:30:01 GST</td></tr><tr><td>ora-tstardb-sys.170930</td><td>ora-tstardb-sys</td><td>2017-09-30 23:30:02 GST</td></tr><tr><td>ora-tstardb-sys.171001</td><td>ora-tstardb-sys</td><td>2017-10-01 23:30:01 GST</td></tr><tr><td>ora-tstardb-sys.171002</td><td>ora-tstardb-sys</td><td>2017-10-02 23:30:02 GST</td></tr><tr><td>ora-tstardb-sys.171003</td><td>ora-tstardb-sys</td><td>2017-10-03 23:30:01 GST</td></tr><tr><td>ora-tstardb-u01.170927</td><td>ora-tstardb-u01</td><td>2017-09-27 23:30:01 GST</td></tr><tr><td>ora-tstardb-u01.170928</td><td>ora-tstardb-u01</td><td>2017-09-28 23:30:01 GST</td></tr><tr><td>ora-tstardb-u01.170929</td><td>ora-tstardb-u01</td><td>2017-09-29 23:30:01 GST</td></tr><tr><td>ora-tstardb-u01.170930</td><td>ora-tstardb-u01</td><td>2017-09-30 23:30:02 GST</td></tr><tr><td>ora-tstardb-u01.171001</td><td>ora-tstardb-u01</td><td>2017-10-01 23:30:01 GST</td></tr><tr><td>ora-tstardb-u01.171002</td><td>ora-tstardb-u01</td><td>2017-10-02 23:30:02 GST</td></tr><tr><td>ora-tstardb-u01.171003</td><td>ora-tstardb-u01</td><td>2017-10-03 23:30:01 GST</td></tr><tr><td>------------------------------------------------------------------</td><td></td><td>  </td></tr></tbody></table></div></div></div></div><div class=\\"col-md-6\\"><div id=\\"oracle-report\\" role=\\"tablist\\" class=\\"panel-group\\"><div class=\\"panel panel-default\\"><div role=\\"tab\\" id=\\"oracle-report-heading-0\\" class=\\"panel-heading\\"><div class=\\"panel-title\\"><a aria-expanded=\\"true\\" class=\\"\\" aria-controls=\\"oracle-report-body-0\\" role=\\"button\\" href=\\"#\\">Database instance info</a></div></div><div class=\\"panel-collapse collapse in\\" id=\\"oracle-report-body-0\\" role=\\"tabpanel\\" aria-labelledby=\\"oracle-report-heading-0\\" aria-expanded=\\"true\\"><div class=\\"panel-body\\"><pre>ORACONF=TSTARDB
DBTYPE=single-ha
DBNAME=TSTARDB
SIDNAME=TSTARDB
DATABASE_CHARACTERSET=AR8MSWIN1256
NODE1=ardb
NODE1IP=192.168.42.120</pre></div></div></div></div></div></div></div></div></div></div>"
`;
